packages/angular/src/generators/scam/scam.ts

```diff
@@ -21,4 +29,20 @@ export async function scamGenerator(tree: Tree, schema: Schema) {
   await formatFiles(tree);
 }
 
+function checkPathUnderProjectRoot(tree: Tree, options: Partial<Schema>) {
+  const project =
+    options.project ?? readWorkspaceConfiguration(tree).defaultProject;
+  const { sourceRoot } = readProjectConfiguration(tree, project);
+
+  if (!options.path) {
+    return;
+  }
+
+  if (!normalize(options.path).startsWith(normalize(sourceRoot))) {
```

Comment:
I think we shouldn't restrict to `sourceRoot` and instead use `root`, it's ok as long as it is inside the root of the project. Also, `sourceRoot` is nullable, it might not be set.

Another things is that we need to use normalize the path differently, just using `normalize` from `path` won't solve the scenario where:

- `project.root` is `apps/app1`
- the provided path is `/apps/app1`

Or vice-versa.

The paths won't match even though they should.

Test cases for these scenarios would be great.

Reply:
that makes sense